x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStep.java

```diff
@@ -41,14 +41,16 @@ public class MountSnapshotStep extends AsyncRetryDuringSnapshotActionStep {
     private final MountSearchableSnapshotRequest.Storage storageType;
     @Nullable
     private final Integer totalShardsPerNode;
+    private final int numberOfReplicas;
 
     public MountSnapshotStep(
         StepKey key,
         StepKey nextStepKey,
         Client client,
         String restoredIndexPrefix,
         MountSearchableSnapshotRequest.Storage storageType,
-        @Nullable Integer totalShardsPerNode
+        @Nullable Integer totalShardsPerNode,
+        int numberOfReplicas
```

Comment:
Can we make this an optional `@Nullable Integer` so that we can treat it as missing in the future? (i.e., if we wanted to change its default value somewhere in the transport action for the mounting, knowing whether it's explicitly 0 or missing is useful)

Reply:
I wonder if we should make it an optional boolean instead? We could conceivably end up not supporting more than 1 replica and instead prefer a time-specification for the replicas here. Narrowing this down to either 0 or 1 replicas could thus maybe help us for the final solution.